### PR TITLE
ci: Only do cron job on main repository

### PR DESCRIPTION
**Pull request by doanac**
I noticed my doanac/meta-qcom-hwe fork was trying to run this cron job every night but fails because there is no runner for it.

### CODE REVIEW

**Review by quaresmajose — APPROVED** [review]
LGTM

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
 
 jobs:
   compile:
+    if: github.repository == 'quic-yocto/meta-qcom-hwe'
```
Comment (ricardosalveti) [inline .github/workflows/nightly-build.yml:11]: Nice, we should have the same logic applied on all jobs, not only nightly.

Comment (doanac) [reply .github/workflows/nightly-build.yml:11]: updated in force push

File: .github/workflows/repolinter.yml [hunk 2]
```diff
@@ -27,4 +28,4 @@ jobs:
         if: steps.check_files.outputs.files_exists == 'false'
         uses: todogroup/repolinter-action@v1
         with:
-          config_url: "https://raw.githubusercontent.com/quic/.github/main/repolint.json"
```
Comment (ndechesne) [inline .github/workflows/repolinter.yml:30]: nitpick.. but unrelated change here.

Comment (doanac) [reply .github/workflows/repolinter.yml]: that was surprisingly hard to fix. my editors fought me. had to fix it Python.

**Comment by ricardosalveti** [comment]
Needs rebase.

**Comment by doanac** [comment]
rebased on latest main

File: .github/workflows/repolinter.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ on:
 
 jobs:
   repolinter:
+    if: github.repository == 'quic-yocto/meta-qcom-hwe'
```
Comment (ndechesne) [inline .github/workflows/repolinter.yml:12]: do we want to leave the jobs which run on github runners, and only limit the ones that run on self hosted runners?

Comment (doanac) [reply .github/workflows/repolinter.yml:12]: I'm fine either way. I got the impression Ricardo was leaning towards this approach

Comment (doanac) [reply .github/workflows/repolinter.yml:12]: @ricardosalveti - whatcha think?

Comment (ricardosalveti) [reply .github/workflows/repolinter.yml:12]: I personally don't like runners to be enabled after creating a fork, creates too much noise. The runs will still be executed on pull requests.

But it might be just me :-)

Comment (ndechesne) [reply .github/workflows/repolinter.yml:12]: ok, fine with me too.

**Review by ricardosalveti — APPROVED** [review]
LGTM
